contrib/format-xml/src/main/java/org/apache/drill/exec/store/xml/XMLMap.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.drill.exec.store.xml;
+
+import org.apache.drill.common.PlanStringBuilder;
+import org.apache.drill.exec.vector.accessor.TupleWriter;
+
+import java.util.Objects;
+
+public class XMLMap {
+
+  private final String mapName;
+  private final int nestingLevel;
```

Comment:
Could you please explain the goal for this field? Currently, it is used only in `equals()` and `toString()` methods.

Reply:
> @cgivre, thanks for the PR, is it possible to incorporate here with schema provisioning schema, so the user will be able to specify how to treat the file for the case of ambiguous files? Also, it may be an additional step in the direction of using an XSD file or link with the schema for XML.

@vvysotskyi 
Thanks for the review.  It is definitely possible and that is my plan.  My thought was to get this committed first w/o the XSD support.  Once this is committed I will add:

* `xsd` support
* Integration with the HTTP REST Storage Plugin so that we can read XML data coming from APIs 
* Support for lists